src/Corax/Queries/SortingMatch.cs

```diff
@@ -135,15 +142,34 @@ public int Fill(Span<long> matches)
 
             Unsafe.SkipInit(out key);
             return false;
-        }
+        }        
 
         [SkipLocalsInit]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private int Fill<W>(Span<long> matches) where W : struct
-        {
-            int take = _take <= 0 ? matches.Length : _take;
+        {             
+            // Important: If you are going to request a massive take like 20K you need to pass at least a 20K size buffer to work with.
+            //            The rationale for such behavior is that sorting has to find among the candidates the order between elements,
+            //            and it can't do so without checking every single element found. If you fail to do so, your results may not be
+            //            correct. 
+            Debug.Assert(_take <= matches.Length);
```

Comment:
That should be a runtime check, no?

Reply:
Another important factor, we can get the total number of entries in the index, and limit to that if the take is higher.

Reply:
I debated myself about that too, and decided against. That is a contract, the query planner must not break the contract, a debit check to catch errors should be enough.